Sources_v3/WebSocketClient/EventMiddlewares/ChannelReadUpdaterMiddleware_Tests.swift

```diff
@@ -0,0 +1,141 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+@testable import StreamChatClient
+import XCTest
+
+class ChannelReadUpdaterMiddleware_Tests: XCTestCase {
+    var middleware: ChannelReadUpdaterMiddleware<DefaultDataTypes>!
+    fileprivate var database: DatabaseContainerMock!
+    
+    override func setUp() {
+        super.setUp()
+        database = try! DatabaseContainerMock(kind: .inMemory)
+        middleware = ChannelReadUpdaterMiddleware(database: database)
+    }
+    
+    func test_messageReadEvent_handledCorrectly() throws {
+        // Save a channel with a channel read
+        let channelId = ChannelId.unique
+        let payload = dummyPayload(with: channelId)
+        
+        assert(payload.channelReads.count == 1)
+        
+        try database.writeSynchronously { (session) in
+            try session.saveChannel(payload: payload)
+        }
+        
+        // Load the channel from the db and check the if fields are correct
+        var loadedChannel: ChannelModel<DefaultDataTypes>? {
+            database.viewContext.loadChannel(cid: channelId)
+        }
+        
+        XCTAssertEqual(loadedChannel?.reads.first?.unreadMessagesCount, 10)
+        XCTAssertEqual(loadedChannel?.reads.first?.lastReadAt, Date(timeIntervalSince1970: 1))
+        
+        let newReadDate = Date(timeIntervalSince1970: 2)
+        let eventPayload = EventPayload<DefaultDataTypes>(
+            eventType: .messageRead,
+            cid: channelId,
+            user: dummyCurrentUser,
+            unreadCount: .noUnread,
+            createdAt: newReadDate
+        )
+        let messageReadEvent = try MessageReadEvent<DefaultDataTypes>(from: eventPayload)
+        
+        // Let the middleware handle the event
+        let handledEvent = try await { middleware.handle(event: messageReadEvent, completion: $0) }
+        
+        XCTAssertEqual(handledEvent?.asEquatable, messageReadEvent.asEquatable)
+        XCTAssertEqual(loadedChannel?.reads.first?.unreadMessagesCount, 0)
+        XCTAssertEqual(loadedChannel?.reads.first?.lastReadAt, newReadDate)
```

Comment:
Interesting that these tests pass when the `write { }` happens asynchronously. Maybe it will fail at least on the CI? :)